net/dns/manager.go

```diff
@@ -257,7 +257,7 @@ func (m *Manager) compileConfig(cfg Config) (rcfg resolver.Config, ocfg OSConfig
 		}
 	}
 
-	if baseCfg == nil || isApple && len(baseCfg.Nameservers) == 0 {
+	if baseCfg == nil {
 		// If there was no base config, or if we're on Apple and the base
 		// config is empty, then we need to fallback to SplitDNS mode.
```

Comment:
```suggestion
		// If there was no base config, then we need to fallback to SplitDNS mode.
```

Reply:
It appears this scenario was already covered by tests; the code we removed wasn't covered because it relied on the value of `GOOS`, which is always Linux in CI, hence no tests ever exercised that branch. As a matter of fact, the tests will fail when running on macOS due to `MatchDomains` getting added to the expected result for some test cases.

I don't know if there's a good way to fake `GOOS` for a given test?

Reply:
Ahh, I forgot we don't have macOS CI runners in this repo :(
Could we make `GOOS` be a field of `dns.Manager`, so that it can be replaced in tests?

Reply:
on it